File: algorithms/bogo_sort/java/BogoSort.java
import java.util.Arrays;
import java.util.List;
import java.util.Collections;

public class BogoSort {
	
	public static List<Integer> bogoSort(List<Integer> nums) {
		if (isSorted(nums)) {
			return nums;
		}
		
		for (;;) {
			Collections.shuffle(nums);
			
			if (isSorted(nums)) {
				break;
			}
		}
		
		return nums;
	}
	
	public static boolean isSorted(List<Integer> x) {
		for (int i = 0; i < x.size() - 1; i++) {
			if (x.get(i) > x.get(i + 1)) {
				return false;
			}
		}
		
		return true;
	}
}

File: algorithms/bogo_sort/ruby/bogo_sort.rb
def bogosort(arr)
  arr.shuffle! while !arr.each_cons(2).all? do |x, y|
    x <= y
  end
end
array = [ 0, 5, 4, 3, 2, 8, 7, 1, 9, 6]
bogosort(array)
puts array # => 0, 1, 2, 3, 4, 5, 6, 7, 8, 9
